### PR TITLE
Removed usage of AspectMock in AnnotationsCheckTest.php

### PR DESCRIPTION
**Pull request by drpayyne**
### Description
Removed usage of AspectMock in `\tests\unit\Magento\FunctionalTestFramework\StaticCheck\AnnotationsCheckTest`

### Fixed Issues (if relevant)

1. https://github.com/magento/magento2/issues/33288: [MFTF] Replace AspectMock with PHPUnit for AnnotationsCheckTest

Resolves https://github.com/magento/magento2/issues/33288

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/verification tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)
 - [x] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Comment by jilu1** [comment]
@drpayyne 
Thank you for your PR!

Same here.  Please ignore the `Error: Resource not accessible by integration` for now.

**Comment by jilu1** [comment]
@magento import pull request to https://github.com/magento-commerce/magento2-functional-testing-framework

**Comment by magento-engcom-team** [comment]
@jilu1 the pull request successfully imported.

File: dev/tests/unit/Magento/FunctionalTestFramework/StaticCheck/AnnotationsCheckTest.php [hunk 1]
```diff
@@ -56,11 +51,9 @@ public function testValidateRequiredAnnotationsNoError()
...
         $expected = [];
 
-        // mock test object
-        $test = AspectMock::double(
-            TestObject::class,
-            ['getAnnotations' => $annotations, 'getName' => 'AnnotationsCheckTest']
-        )->make();
+        $test = $this->createMock(TestObject::class);
+
+        $test->expects($this->once())->method('getAnnotations')->willReturn($annotations);
```
Comment (jilu1) [inline dev/tests/unit/Magento/FunctionalTestFramework/StaticCheck/AnnotationsCheckTest.php:56]: Can we also add `$test->expects($this->once())->method('getName')->willReturn('AnnotationsCheckTest');`?

Comment (jilu1) [reply dev/tests/unit/Magento/FunctionalTestFramework/StaticCheck/AnnotationsCheckTest.php:56]: Never mind. Actually getName() is not invoked in this test case.

Comment (drpayyne) [reply dev/tests/unit/Magento/FunctionalTestFramework/StaticCheck/AnnotationsCheckTest.php:56]: Yes, I added that expectation with a `once` and the test failed, that's why I removed it. :)

**Pull request merged**